ring/partition_ring_watcher.go

```diff
@@ -46,6 +49,16 @@ func NewPartitionRingWatcher(name, key string, kv kv.Client, logger log.Logger,
 			Help:        "Number of partitions by state in the partitions ring.",
 			ConstLabels: map[string]string{"name": name},
 		}, []string{"state"}),
+		partitionLockStateGaugeVec: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+			Name:        "partition_ring_partition_state_change_locked",
+			Help:        "Whether the partition state change is locked. 1 if locked, 0 if unlocked.",
+			ConstLabels: map[string]string{"name": name},
+		}, []string{"partition_id", "state"}),
+		partitionLockTimestampGaugeVec: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+			Name:        "partition_ring_partition_state_change_locked_timestamp_seconds",
+			Help:        "Unix timestamp (seconds) of when the partition state change lock was last modified. 0 if unlocked.",
+			ConstLabels: map[string]string{"name": name},
+		}, []string{"partition_id", "state"}),
```

Comment:
These metrics are going to be high cardinality because exposed for each partition by each pod. Are they really required? Looks like useful only once in a lifetime (maybe when investigating an incident) and maybe can be replaced by a log when the lock change? It could be logged here by the watcher.